Source/Engine/Physics/Joints/Joint.cpp

```diff
@@ -16,10 +19,111 @@ Joint::Joint(const SpawnParams& params)
     , _joint(nullptr)
     , _breakForce(MAX_float)
     , _breakTorque(MAX_float)
-    , _targetAnchor(Vector3::Zero)
-    , _targetAnchorRotation(Quaternion::Identity)
+    , LocalPoseActor0()
+    , LocalPoseActor1()
 {
-    Target.Changed.Bind<Joint, &Joint::OnTargetChanged>(this);
+    Actor0.Changed.Bind<Joint, &Joint::UpdateJointActors>(this);
+    Actor1.Changed.Bind<Joint, &Joint::UpdateJointActors>(this);
+}
+
+void Joint::SetActors(RigidBody* Actor0, RigidBody* Actor1)
+{
+    if (_joint)
+    {
+        PhysicsBackend::SetJointActors
+        (
+            _joint,
+            Actor0 ? Actor0->GetPhysicsActor() : nullptr,
+            Actor1 ? Actor1->GetPhysicsActor() : nullptr
+        );
+    }
+}
+#pragma region SheredCode
+#define ImplementSheredCodeForActor(N)\
```

Comment:
is this suppose to be Shared or sheared?

Reply:
o ye 😂 compiler will not care
but it can be changed